### DISCUSSION TITLE
Is there any way of using an asyncronous filter function in the SpotlightProvider

### DISCUSSION THREAD

**Discussion by valerius21** [mantinedev/mantine#2253] [Q&A] [answered]
Hi there,

I'm working on a filter function that involves some asynchronous function calls. Reading the documentation, only non-Promise functions are supported.

Are there any workarounds?

Cheers!

**Comment by rtivital** [comment, marked as answer]
Spotlight and other similar components only support sync operations, you need to resolve async logic on your side and provide prepared data to components.

**Reply by viveknigam3003** [reply]
Is there any example for handling async operations with spotlight?
